liblumen_alloc/src/erts/term/list.rs

```diff
@@ -355,6 +357,12 @@ impl Iterator for Iter {
     }
 }
 
+#[derive(Debug, Error)]
```

Comment:
Again, I'd just make this a struct called `ImproperListError` which can be used anywhere that a proper list is expected but not given. Having function-specific exceptions is too granular, and will result in a lot of duplication.